index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
```

Comment:
> Not sure if that's the right way of thinking about it though.

I don't think this is the right way to think about it. We very much want to ensure that the values we return are members of a concrete type. The type will look *something* like this:

```haskell
{ namespace :: Nullable String, name :: String, version :: Nullable String }
```

Now, we could use `TypeIdentifier` as a synonym for this record type, but it's probably clearer to use the record type itself in the type signature.

> ```javascript
> type(x).version < type(y).version
> ```

This seems to work correctly because `null` is coerced to `0`. If we want to support this operation (I think we do) we should make the `version` type `Number` rather than `Nullable String` or `Nullable Number`, and default to `0` if no version is specified.